File: capture_screens.py
from playwright.sync_api import sync_playwright
import os, time

BASE = "http://localhost:4600/ui_design"
OUT  = os.path.join(os.path.dirname(__file__), "screenshots")
os.makedirs(OUT, exist_ok=True)

SCREENS = [
    ("login",            None),
    ("dashboard",        "showApp(); showScreen('dashboard');"),
    ("clients",          "showApp(); showScreen('clients');"),
    ("client-detail",    "showApp(); showScreen('client-detail');"),
    ("jobs",             "showApp(); showScreen('jobs');"),
    ("job-detail",       "showApp(); showScreen('job-detail');"),
    ("pipeline",         "showApp(); showScreen('pipeline');"),
    ("candidates",       "showApp(); showScreen('candidates');"),
    ("candidate-profile","showApp(); showScreen('candidate-profile');"),
    ("upload-cv",        "showApp(); showScreen('upload-cv');"),
    ("cv-review",        "showApp(); showScreen('cv-review');"),
    ("cv-template",      "showApp(); showScreen('cv-template');"),
    ("cv-download",      "showApp(); showScreen('cv-download');"),
    ("activity",         "showApp(); showScreen('activity');"),
    ("reminders",        "showApp(); showScreen('reminders');"),
    ("files-root",       "showApp(); showScreen('files'); showFWRoot();"),
    ("files-entity",     "showApp(); showScreen('files'); showFWEntity();"),
    ("files-folder",     "showApp(); showScreen('files'); showFWFolder();"),
    ("onboarding",       "showApp(); showScreen('onboarding');"),
]

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={"width": 1280, "height": 820})

    for name, js in SCREENS:
        page.goto(BASE, wait_until="networkidle")
        if js:
            page.evaluate(js)
        page.wait_for_timeout(400)
        path = os.path.join(OUT, f"{name}.png")
        page.screenshot(path=path, full_page=False)
        print(f"  OK {name}")

    browser.close()
    print(f"\nDone — screenshots in: {OUT}")

File: capture_screens.js
const puppeteer = require('puppeteer');
const path = require('path');
const fs = require('fs');

const BASE = 'http://localhost:4600/ui_design';
const OUT  = path.join(__dirname, 'screenshots');

if (!fs.existsSync(OUT)) fs.mkdirSync(OUT);

const SCREENS = [
  { name: 'login',             fn: async (page) => {
      // login is shown by default on load
    }
  },
  { name: 'dashboard',         fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('dashboard'); }); } },
  { name: 'pipeline',          fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('pipeline'); }); } },
  { name: 'clients',           fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('clients'); }); } },
  { name: 'client-detail',     fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('client-detail'); }); } },
  { name: 'jobs',              fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('jobs'); }); } },
  { name: 'job-detail',        fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('job-detail'); }); } },
  { name: 'candidates',        fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('candidates'); }); } },
  { name: 'candidate-profile', fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('candidate-profile'); }); } },
  { name: 'upload-cv',         fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('upload-cv'); }); } },
  { name: 'cv-review',         fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('cv-review'); }); } },
  { name: 'cv-template',       fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('cv-template'); }); } },
  { name: 'cv-download',       fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('cv-download'); }); } },
  { name: 'activity',          fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('activity'); }); } },
  { name: 'reminders',         fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('reminders'); }); } },
  { name: 'files-root',        fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('files'); showFWRoot(); }); } },
  { name: 'files-entity',      fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('files'); showFWEntity(); }); } },
  { name: 'files-folder',      fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('files'); showFWFolder(); }); } },
  { name: 'onboarding',        fn: async (page) => { await page.evaluate(() => { showApp(); showScreen('onboarding'); }); } },
];

(async () => {
  const browser = await puppeteer.launch({ headless: true, args: ['--no-sandbox'] });
  const page = await browser.newPage();
  await page.setViewport({ width: 1280, height: 800, deviceScaleFactor: 1.5 });

  for (const screen of SCREENS) {
    await page.goto(BASE, { waitUntil: 'networkidle0' });
    await screen.fn(page);
    await new Promise(r => setTimeout(r, 300));
    const file = path.join(OUT, `${screen.name}.png`);
    await page.screenshot({ path: file, fullPage: false });
    console.log(`✓ ${screen.name}`);
  }

  await browser.close();
  console.log('All screenshots saved to', OUT);
})();
